services/painting-process-equipment-defect-detection-model-service/app/dependencies.py

```
@@ -0,0 +1,99 @@
+    and ensures global state is consistent even on failure.
+    """
+    global _model, _explainer, _config, _initialization_complete
+
+    # 1. 설정 파일 로드
+    try:
+        with open(config_path, 'r', encoding='utf-8') as f:
+            _config = yaml.safe_load(f)
+        print("설정 파일 로드 완료")
+    except FileNotFoundError:
+        print(f"오류: 설정 파일을 찾을 수 없습니다. 경로를 확인해주세요: {config_path}")
+        raise FileNotFoundError(f"Configuration file not found at {config_path}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)